debugger/src/debug_adapter/protocol.rs

```diff
@@ -63,11 +63,21 @@ impl<R: Read, W: Write> DapAdapter<R, W> {
         let response_header = format!("Content-Length: {}\r\n\r\n", response_body.len());
 
         self.output.write_all(response_header.as_bytes())?;
+        self.output.flush()?;
 
-        if let Err(err) = self.output.write_all(response_body) {
-            // TODO: Implement catch of errorkind "WouldBlock" and retry
-            tracing::error!("send_data - body: {:?}", err);
-            return Err(err);
+        // NOTE: Sometimes when writing large response, the debugger will fail with an IO error (ErrorKind::WouldBlock == error.kind())
+        // Retrying is not an option, because there is no way to know how much data was already written.
+        // Trying to resend smaller chunks after this error just causes the request to fail.
+        // The strategy then is to avoid this error as much as possible, by always sending data in smaller chunks.
+        if response_body.len() > 1024 {
+            let chunked_output = response_body.chunks(1024);
+            for chunk in chunked_output {
+                self.output.write_all(chunk)?;
+                self.output.flush()?;
```

Comment:
This shouldn't be necessary here, since a flush is done at the end.

Reply:
Please see my comment above. I only added these flush() calls because it breaks consistently without them.